api/server/api.cc

```diff
@@ -302,6 +338,7 @@ bool API::action_sg_add(const app::Sg &sg, app::Error *error) {
     }
 
     sg_update(sg);
+    sg_update_rule_members(sg);
```

Comment:
can't you call sg_update_rule_members at the end of sg_update instead of calling it after each sg_update ?